### PR TITLE
[#1847693] replace all `log` with `AzureMessager` in azure-toolkit-libs

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
[AB#1847693](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1847693)

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/CommandHandlerImpl.java [hunk 1]
```diff
@@ -62,7 +64,7 @@ protected String getOutputFromProcess(final Process process) throws IOException
     protected Process runCommand(final String command,
                                  final boolean showStdout,
                                  final String workingDirectory) throws IOException, InterruptedException {
-        Log.debug("Executing command: " + StringUtils.join(command, " "));
+        log.debug("Executing command: " + StringUtils.join(command, " "));
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/CommandHandlerImpl.java:67]: replace with `AzureMessager.getMessager().log`?

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/FunctionCoreToolsHandlerImpl.java [hunk 2]
```diff
@@ -57,11 +58,12 @@ protected void assureRequirementAddressed() throws AzureExecutionException {
         if (localVersion == null || LEAST_SUPPORTED_VERSION.greaterThan(Version.valueOf(localVersion))) {
             throw new AzureExecutionException(CANNOT_AUTO_INSTALL);
         }
+        final IAzureMessager messager = AzureMessager.getMessager();
         // Verify whether local function core tools is the latest version
         if (latestCoreVersion == null) {
-            Log.warn(GET_LATEST_VERSION_FAIL);
+            messager.warning(GET_LATEST_VERSION_FAIL);
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/FunctionCoreToolsHandlerImpl.java:64]: may make messager as a member variable

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/CommandHandlerImpl.java [hunk 3]
```diff
@@ -94,19 +96,19 @@ protected void handleExitValue(int exitValue,
...
                                    final InputStream inputStream) throws AzureExecutionException, IOException {
-        Log.debug("Process exit value: " + exitValue);
+        log.debug("Process exit value: " + exitValue);
         if (!validReturnCodes.contains(Integer.toUnsignedLong(exitValue))) {
             // input stream is a merge of standard output and standard error of the sub-process
             showErrorIfAny(inputStream);
-            Log.error(errorMessage);
+            AzureMessager.getMessager().error(errorMessage);
             throw new AzureExecutionException(errorMessage);
         }
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/CommandHandlerImpl.java:105]: how do you differentiate to use "log" and "messager" ?

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/CommandHandlerImpl.java (hunk 1 above)
Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/CommandHandlerImpl.java:67]: debug message should not show to user. so debug should not be part of AzureMessager.

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/FunctionCoreToolsHandlerImpl.java (hunk 2 above)
Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/handlers/FunctionCoreToolsHandlerImpl.java:64]: no, messager may change to the runtime context.

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/appservice/handlers/artifact/FTPUploader.java [hunk 4]
```diff
@@ -62,49 +58,40 @@ public void uploadDirectoryWithRetries(final String ftpServer, final String user
...
-     * @param password
-     * @param sourceDirectoryPath
-     * @param targetDirectoryPath
      * @return Boolean to indicate whether uploading is successful.
      */
     protected boolean uploadDirectory(final String ftpServer, final String username, final String password,
                                       final String sourceDirectoryPath, final String targetDirectoryPath) {
-        Log.debug("FTP username: " + username);
+        final IAzureMessager messager = AzureMessager.getMessager();
+        log.debug("FTP username: " + username);
```
[the remark this answers is not in the document]
Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/appservice/handlers/artifact/FTPUploader.java:66]: debug message should not show to user. so debug should not be part of AzureMessager.
IMO, AzureMessager only responsible for showing message to user.

**Review by andxu — DISMISSED** [review]

**Pull request merged**